<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />


<meta name="date" content="2025-08-09" />

<title>TCGA-BRCA Gene Expression Analysis with bregr</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">TCGA-BRCA Gene Expression Analysis with
bregr</h1>
<h4 class="author">Shixiang Wang</h4>
<address class="author_afil">
Central South University<br><a class="author_email" href="mailto:#"><a href="mailto:wangshx@csu.edu.cn" class="email">wangshx@csu.edu.cn</a></a>
</address>
<h4 class="date">2025-08-09</h4>



<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co">#remotes::install_github(&quot;wanglabcsu/bregr&quot;, dependencies = TRUE)</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">library</span>(bregr)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="co">#&gt; Welcome to &#39;bregr&#39; package!</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="co">#&gt; =======================================================================</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="co">#&gt; You are using bregr version 1.0.0.9000</span></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="co">#&gt; Project home : https://github.com/WangLabCSU/bregr</span></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a><span class="co">#&gt; Documentation: https://wanglabcsu.github.io/bregr/</span></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a><span class="co">#&gt; Cite as      : arXiv:2110.14232</span></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="co">#&gt; =======================================================================</span></span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a><span class="fu">library</span>(UCSCXenaTools)</span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a><span class="co">#&gt; =========================================================================================</span></span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a><span class="co">#&gt; UCSCXenaTools version 1.6.0</span></span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a><span class="co">#&gt; Project URL: https://github.com/ropensci/UCSCXenaTools</span></span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a><span class="co">#&gt; Usages: https://cran.r-project.org/web/packages/UCSCXenaTools/vignettes/USCSXenaTools.html</span></span>
<span id="cb1-18"><a href="#cb1-18" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-19"><a href="#cb1-19" tabindex="-1"></a><span class="co">#&gt; If you use it in published research, please cite:</span></span>
<span id="cb1-20"><a href="#cb1-20" tabindex="-1"></a><span class="co">#&gt; Wang et al., (2019). The UCSCXenaTools R package: a toolkit for accessing genomics data</span></span>
<span id="cb1-21"><a href="#cb1-21" tabindex="-1"></a><span class="co">#&gt;   from UCSC Xena platform, from cancer multi-omics to single-cell RNA-seq.</span></span>
<span id="cb1-22"><a href="#cb1-22" tabindex="-1"></a><span class="co">#&gt;   Journal of Open Source Software, 4(40), 1627, https://doi.org/10.21105/joss.01627</span></span>
<span id="cb1-23"><a href="#cb1-23" tabindex="-1"></a><span class="co">#&gt; =========================================================================================</span></span>
<span id="cb1-24"><a href="#cb1-24" tabindex="-1"></a><span class="co">#&gt;                               --Enjoy it--</span></span>
<span id="cb1-25"><a href="#cb1-25" tabindex="-1"></a><span class="fu">library</span>(UCSCXenaShiny)</span>
<span id="cb1-26"><a href="#cb1-26" tabindex="-1"></a><span class="co">#&gt; =========================================================================================</span></span>
<span id="cb1-27"><a href="#cb1-27" tabindex="-1"></a><span class="co">#&gt; UCSCXenaShiny version 2.2.0</span></span>
<span id="cb1-28"><a href="#cb1-28" tabindex="-1"></a><span class="co">#&gt; Project URL: https://github.com/openbiox/UCSCXenaShiny</span></span>
<span id="cb1-29"><a href="#cb1-29" tabindex="-1"></a><span class="co">#&gt; Usages: https://openbiox.github.io/UCSCXenaShiny/</span></span>
<span id="cb1-30"><a href="#cb1-30" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-31"><a href="#cb1-31" tabindex="-1"></a><span class="co">#&gt; If you use it in published research, please cite:</span></span>
<span id="cb1-32"><a href="#cb1-32" tabindex="-1"></a><span class="co">#&gt;   Shensuo Li, Yuzhong Peng, Minjun Chen, Yankun Zhao, Yi Xiong, Jianfeng Li, Peng Luo, </span></span>
<span id="cb1-33"><a href="#cb1-33" tabindex="-1"></a><span class="co">#&gt;   Haitao Wang, Fei Zhao, Qi Zhao, Yanru Cui, Sujun Chen, Jian-Guo Zhou, Shixiang Wang,  </span></span>
<span id="cb1-34"><a href="#cb1-34" tabindex="-1"></a><span class="co">#&gt;   Facilitating integrative and personalized oncology omics analysis with UCSCXenaShiny, </span></span>
<span id="cb1-35"><a href="#cb1-35" tabindex="-1"></a><span class="co">#&gt;   Communications Biology, 1200 (2024),  https://doi.org/10.1038/s42003-024-06891-2</span></span>
<span id="cb1-36"><a href="#cb1-36" tabindex="-1"></a><span class="co">#&gt; =========================================================================================</span></span>
<span id="cb1-37"><a href="#cb1-37" tabindex="-1"></a><span class="co">#&gt;                               --Enjoy it--</span></span></code></pre></div>
<div id="download-data" class="section level2">
<h2>Download Data</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">XenaGenerate</span>() <span class="sc">|&gt;</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>  <span class="fu">XenaFilter</span>(<span class="at">filterDatasets =</span> <span class="st">&quot;TCGA.BRCA.sampleMap/HiSeqV2_PANCAN&quot;</span>)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="co">#&gt; class: XenaHub </span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="co">#&gt; hosts():</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="co">#&gt;   https://tcga.xenahubs.net</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="co">#&gt; cohorts() (1 total):</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="co">#&gt;   TCGA Breast Cancer (BRCA)</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="co">#&gt; datasets() (1 total):</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="co">#&gt;   TCGA.BRCA.sampleMap/HiSeqV2_PANCAN</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">XenaGenerate</span>() <span class="sc">|&gt;</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>  <span class="fu">XenaFilter</span>(<span class="at">filterDatasets =</span> <span class="st">&quot;TCGA.BRCA.sampleMap/HiSeqV2_PANCAN&quot;</span>) <span class="sc">|&gt;</span> </span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>  <span class="fu">XenaQuery</span>() <span class="sc">|&gt;</span> </span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>  <span class="fu">XenaDownload</span>(<span class="at">destdir =</span> <span class="st">&quot;&quot;</span>)</span></code></pre></div>
</div>
<div id="preprocessing-data" class="section level2">
<h2>Preprocessing Data</h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>data <span class="ot">=</span> <span class="fu">XenaPrepare</span>(<span class="st">&quot;HiSeqV2_PANCAN.gz&quot;</span>)</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="fu">head</span>(data[, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>])</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="co">#&gt; # A tibble: 6 × 10</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="co">#&gt;   sample `TCGA-AR-A5QQ-01` `TCGA-D8-A1JA-01` `TCGA-BH-A0BQ-01` `TCGA-BH-A0BT-01`</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;              &lt;dbl&gt;             &lt;dbl&gt;             &lt;dbl&gt;             &lt;dbl&gt;</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a><span class="co">#&gt; 1 ARHGE…            -0.357           -2.43             -0.542            -0.844 </span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a><span class="co">#&gt; 2 HIF3A             -3.98            -1.89             -2.83             -4.21  </span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a><span class="co">#&gt; 3 RNF17             -0.531            0.0935            0.0216           -0.531 </span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a><span class="co">#&gt; 4 RNF10             -0.410            0.141             0.189             1.41  </span></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a><span class="co">#&gt; 5 RNF11              0.161            2.56              0.443             0.123 </span></span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a><span class="co">#&gt; 6 RNF13             -0.518            0.380            -0.0497           -0.0659</span></span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a><span class="co">#&gt; # ℹ 5 more variables: `TCGA-A8-A06X-01` &lt;dbl&gt;, `TCGA-A8-A096-01` &lt;dbl&gt;,</span></span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a><span class="co">#&gt; #   `TCGA-BH-A0C7-01` &lt;dbl&gt;, `TCGA-AC-A5XU-01` &lt;dbl&gt;, `TCGA-PE-A5DE-01` &lt;dbl&gt;</span></span></code></pre></div>
<p>Firstly, we need to convert genes to columns.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>data_t <span class="ot">=</span> tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(data, <span class="at">cols =</span> <span class="fu">colnames</span>(data)[<span class="sc">-</span><span class="dv">1</span>]) <span class="sc">|&gt;</span> </span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">id_cols =</span> <span class="st">&quot;name&quot;</span>, <span class="at">names_from =</span> <span class="st">&quot;sample&quot;</span>, <span class="at">values_from =</span> <span class="st">&quot;value&quot;</span>, <span class="at">values_fill =</span> <span class="cn">NA_real_</span>)</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="fu">head</span>(data_t[, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>])</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="co">#&gt; # A tibble: 6 × 10</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="co">#&gt;   name     ARHGEF10L HIF3A   RNF17   RNF10  RNF11   RNF13 GTF2IP1    REM1  MTVR2</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;        &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt;</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="co">#&gt; 1 TCGA-AR…   -0.357  -3.98 -0.531  -0.410   0.161 -0.518  -0.896  -1.02   -0.423</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a><span class="co">#&gt; 2 TCGA-D8…   -2.43   -1.89  0.0935  0.141   2.56   0.380  -0.188  -1.71   -0.423</span></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a><span class="co">#&gt; 3 TCGA-BH…   -0.542  -2.83  0.0216  0.189   0.443 -0.0497  0.326   0.449  -0.423</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a><span class="co">#&gt; 4 TCGA-BH…   -0.844  -4.21 -0.531   1.41    0.123 -0.0659  0.150  -1.51   -0.423</span></span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a><span class="co">#&gt; 5 TCGA-A8…   -0.222  -4.97 -0.531   0.226   0.287 -0.342   0.169   0.0139  1.38 </span></span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a><span class="co">#&gt; 6 TCGA-A8…   -0.0975 -5.28  0.345   0.0342 -0.112 -0.0621 -0.0119  0.0606 -0.423</span></span></code></pre></div>
<p>Next join the overall survival data.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>data2 <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">inner_join</span>(</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>  tcga_surv <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">select</span>(sample, OS, OS.time) <span class="sc">|&gt;</span> </span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(OS), <span class="sc">!</span><span class="fu">is.na</span>(OS.time)),</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>  data_t, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">&quot;sample&quot;</span> <span class="ot">=</span> <span class="st">&quot;name&quot;</span>)</span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>)</span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a><span class="fu">head</span>(data2[, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>])</span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a><span class="co">#&gt;            sample OS OS.time  ARHGEF10L     HIF3A       RNF17       RNF10</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a><span class="co">#&gt; 1 TCGA-3C-AAAU-01  0    4047  0.6073075 -3.194126 -0.53103501 -0.14587199</span></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a><span class="co">#&gt; 2 TCGA-3C-AALI-01  0    4005 -0.6411925 -4.928226  0.09546499  0.09812801</span></span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a><span class="co">#&gt; 3 TCGA-3C-AALJ-01  0    1474  1.0828075 -4.623726 -0.53103501  0.48402801</span></span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a><span class="co">#&gt; 4 TCGA-3C-AALK-01  0    1448  0.1216075 -2.881526 -0.53103501  0.17912801</span></span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a><span class="co">#&gt; 5 TCGA-4H-AAAK-01  0     348  0.4202075 -3.282726 -0.53103501 -0.02097199</span></span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a><span class="co">#&gt; 6 TCGA-5L-AAT0-01  0    1477  0.4748075 -2.775226 -0.53103501  0.24602801</span></span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a><span class="co">#&gt;         RNF11      RNF13     GTF2IP1</span></span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a><span class="co">#&gt; 1  0.23742187 -0.2992099 -0.14269449</span></span>
<span id="cb6-16"><a href="#cb6-16" tabindex="-1"></a><span class="co">#&gt; 2 -0.54197813 -0.3229099 -0.04469449</span></span>
<span id="cb6-17"><a href="#cb6-17" tabindex="-1"></a><span class="co">#&gt; 3 -0.18367813 -0.9190099  0.26110551</span></span>
<span id="cb6-18"><a href="#cb6-18" tabindex="-1"></a><span class="co">#&gt; 4  0.03922187 -0.4549099  0.18030551</span></span>
<span id="cb6-19"><a href="#cb6-19" tabindex="-1"></a><span class="co">#&gt; 5 -0.11797813 -0.5578099 -0.17379449</span></span>
<span id="cb6-20"><a href="#cb6-20" tabindex="-1"></a><span class="co">#&gt; 6 -0.37837813 -0.8308099  0.33890551</span></span></code></pre></div>
<p>So we next run <strong>bregr</strong> with 1210 samples, 2.053^{4}
variables.</p>
</div>
<div id="execute-bregr-pipeline" class="section level2">
<h2>Execute bregr Pipeline</h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>gene_names <span class="ot">=</span> <span class="fu">names</span>(data_t)[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="co"># options(</span></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="co">#   bregr.save_model = TRUE,</span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="co">#   bregr.path = &quot;./bregr_models&quot;</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a><span class="co"># )</span></span></code></pre></div>
<p>With parallel computation:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="fu">system.time</span>(</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>  <span class="fu">br_pipeline</span>(</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>    data2,</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">c</span>(<span class="st">&quot;OS.time&quot;</span>, <span class="st">&quot;OS&quot;</span>),</span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>    <span class="at">x =</span> gene_names, </span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">&quot;coxph&quot;</span>,</span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>    <span class="at">run_parallel =</span> <span class="dv">40</span></span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a>    <span class="fu">suppressWarnings</span>()</span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>)</span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a><span class="co">#&gt; exponentiate estimates of model(s) constructed from coxph method at default</span></span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a><span class="co">#&gt; ■■■                                5% | ETA: 22s</span></span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a><span class="co">#&gt; ■■■■■■■■■                         26% | ETA: 13s</span></span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-16"><a href="#cb8-16" tabindex="-1"></a><span class="co">#&gt; ■■■■■■■■■■■■■■■                   47% | ETA:  8s</span></span>
<span id="cb8-17"><a href="#cb8-17" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-18"><a href="#cb8-18" tabindex="-1"></a><span class="co">#&gt; ■■■■■■■■■■■■■■■■■■■               60% | ETA:  7s</span></span>
<span id="cb8-19"><a href="#cb8-19" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-20"><a href="#cb8-20" tabindex="-1"></a><span class="co">#&gt; ■■■■■■■■■■■■■■■■■■■■■■■■          78% | ETA:  4s</span></span>
<span id="cb8-21"><a href="#cb8-21" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-22"><a href="#cb8-22" tabindex="-1"></a><span class="co">#&gt; ■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■    98% | ETA:  0s</span></span>
<span id="cb8-23"><a href="#cb8-23" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-24"><a href="#cb8-24" tabindex="-1"></a><span class="co">#&gt; ■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■  100% | ETA:  0s</span></span>
<span id="cb8-25"><a href="#cb8-25" tabindex="-1"></a><span class="co">#&gt;     user   system  elapsed </span></span>
<span id="cb8-26"><a href="#cb8-26" tabindex="-1"></a><span class="co">#&gt; 3295.860 1466.084 3821.251</span></span></code></pre></div>
<p>Without parallel computation:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="fu">system.time</span>(</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>  <span class="fu">br_pipeline</span>(</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>    data2,</span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">c</span>(<span class="st">&quot;OS.time&quot;</span>, <span class="st">&quot;OS&quot;</span>),</span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>    <span class="at">x =</span> gene_names, </span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">&quot;coxph&quot;</span></span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a>    <span class="fu">suppressWarnings</span>()</span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a>)</span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a><span class="co">#&gt; exponentiate estimates of model(s) constructed from coxph method at default</span></span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a><span class="co">#&gt;      user    system   elapsed </span></span>
<span id="cb9-12"><a href="#cb9-12" tabindex="-1"></a><span class="co">#&gt; 15349.575     6.829 15418.882</span></span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
